utilities/backupable/backupable_db_test.cc

```diff
@@ -2618,6 +2620,176 @@ TEST_P(BackupEngineRateLimitingTestWithParam, RateLimiting) {
 
   AssertBackupConsistency(0, 0, 100000, 100010);
 }
+
+TEST_P(BackupEngineRateLimitingTestWithParam, RateLimitingVerifyBackup) {
+  const std::size_t kMicrosPerSec = 1000 * 1000LL;
+  std::shared_ptr<RateLimiter> backupThrottler(NewGenericRateLimiter(
+      1, 100 * 1000 /* refill_period_us */, 10 /* fairness */,
+      RateLimiter::Mode::kAllIo /* mode */));
+
+  bool makeThrottler = std::get<0>(GetParam());
+  if (makeThrottler) {
+    backupable_options_->backup_rate_limiter = backupThrottler;
+  }
+
+  bool is_single_threaded = std::get<1>(GetParam()) == 0 ? true : false;
+  backupable_options_->max_background_operations = is_single_threaded ? 1 : 10;
+
+  const std::uint64_t backup_rate_limiter_limit = std::get<2>(GetParam()).first;
+  if (makeThrottler) {
+    backupable_options_->backup_rate_limiter->SetBytesPerSecond(
+        backup_rate_limiter_limit);
+  } else {
+    backupable_options_->backup_rate_limit = backup_rate_limiter_limit;
+  }
+
+  DestroyDB(dbname_, Options());
+  OpenDBAndBackupEngine(true /* destroy_old_data */);
+  FillDB(db_.get(), 0, 100000);
+  ASSERT_OK(backup_engine_->CreateNewBackup(db_.get(),
+                                            false /* flush_before_backup */));
+
+  std::vector<BackupInfo> backup_infos;
+  BackupInfo backup_info;
+  backup_engine_->GetBackupInfo(&backup_infos);
+  ASSERT_EQ(1, backup_infos.size());
+  const int backup_id = 1;
+  ASSERT_EQ(backup_id, backup_infos[0].backup_id);
+  ASSERT_OK(backup_engine_->GetBackupInfo(backup_id, &backup_info,
+                                          true /* include_file_details */));
+
+  std::uint64_t bytes_read_during_verify_backup = 0;
+  for (BackupFileInfo backup_file_info : backup_info.file_details) {
+    bytes_read_during_verify_backup += backup_file_info.size;
+  }
+
+  auto start_verify_backup = db_chroot_env_->NowMicros();
+  ASSERT_OK(
+      backup_engine_->VerifyBackup(backup_id, true /* verify_with_checksum */));
+  auto verify_backup_time = db_chroot_env_->NowMicros() - start_verify_backup;
+  auto rate_limited_verify_backup_time =
+      (bytes_read_during_verify_backup * kMicrosPerSec) /
+      backup_rate_limiter_limit;
+
+  if (makeThrottler) {
+    EXPECT_GE(verify_backup_time, 0.8 * rate_limited_verify_backup_time);
+  } else {
+    EXPECT_LT(verify_backup_time, rate_limited_verify_backup_time);
```

Comment:
Unfortunately, we cannot have tests that check execution time is below some threshold, because this will be flaky. At least remove that part.

These rate limiting tests should really be upgraded to use something other than elapsed clock time for verification, especially since these new tests will likely add quite a bit of running time to the backup tests. They could use either GetTotalRequests() or mock time (time_elapse_only_sleep etc.) as in RateLimiterTest.AutoTuneIncreaseWhenFull. One challenge is when specifying just a rate, the limiter object is hidden from view, but it should be possible to make it accessible to the running test.

But improving the rate limiter test speed and accuracy could be a follow-up PR.

Reply:
> One challenge is when specifying just a rate, the limiter object is hidden from view, but it should be possible to make it accessible to the running test.

Yes - that was the main challenge I had when implementing this test and I guess we can do something like sync point to make it accessible.

> These rate limiting tests should really be upgraded to use something other than elapsed clock time for verification, especially since these new tests will likely add quite a bit of running time to the backup tests. 

Opened a task for this.

Reply:
Offline discussion: we will need something to check the existing behavior of VerifyBackup is not affected and this can be included in the follow-up improvement PR mentioned above.